test/trackForMutations.spec.js

```diff
@@ -18,7 +18,27 @@ describe('trackForMutations', () => {
   function testCasesForNonMutation(spec) {
     it('returns false', () => {
       const state = spec.getState();
-      const tracker = trackForMutations(isImmutable, state);
+      const tracker = trackForMutations(isImmutable, undefined, state);
+      const newState = spec.fn(state);
+
+      expect(
+        tracker.detectMutations()
+      ).toEqual({wasMutated: false});
+    });
+  }
+
+  function testCasesForIgnoreMutationDetectionByPath(spec) {
```

Comment:
I'd rather keep the same mutation and non mutation groups and allow to specify the middleware options in each of the specs, something like:

```js
const spec = {
  middlewareOptions: {
    ignore: [...]
  },
  getState: () => ({ a:1, b:2 }),
  fn: (s) => s
};
```